src/request.ts

```diff
@@ -17,13 +17,31 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import axios, { AxiosRequestConfig } from 'axios';
+import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
+import { getHttpAgents } from './http-agent';
+import { ScannerProperties, ScannerProperty } from './types';
 
-export function fetch(token: string, config: AxiosRequestConfig) {
-  return axios({
-    headers: {
-      Authorization: `Bearer ${token}`,
-    },
-    ...config,
-  });
+// The axios instance is private to this module
+let _axiosInstance: AxiosInstance | null = null;
+
+export function initializeAxios(properties: ScannerProperties) {
+  const token = properties[ScannerProperty.SonarToken];
+  const agents = getHttpAgents(properties);
+
+  if (!_axiosInstance) {
+    _axiosInstance = axios.create({
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+      ...agents,
+    });
```

Comment:
Agreed, but I didn't do it because in this PR we have to make a `fetch(..)` to [another domain](https://github.com/SonarSource/sonar-scanner-npm/pull/117/files#diff-81ff4ca345b84f8478fd75a7d8b9520708b0a5243faa5abb3a2821586931bc0aR73), so not sure that'd work with that, wdyt?

Reply:
I would just override the baseUrl for this specific scenario, it will also make it obvious that it is a different domain

I think we can change `download` to instead of taking a URL, to take an Axios Config Object where we can override more values if needed.